### ISSUE TITLE
How caching works for StringSignature using Glide v3

### ISSUE THREAD

**Issue by KPangy** [bumptech/glide#3946]
I have a caching mechanism for every 24 hours to update the StringSignature to ensure a new URL image is downloaded and rendered. DiskStrategy is set to ALL.

If I have a previously cached image and 24+ hours has passed, I update the StringSignature and make the call to update image but it fails for whatever reason. Will the previously cached image be displayed (this is the requirement) or will placeholder be displayed?

Basically I'd like the previously cached image be the placeholder if StringSignature has been changed. Please advise.

**Comment by sjudd** [comment]
In Glide v4 you can use .error() to start a request with the previous signature only if the new request fails.

In v3 you can probably make the same pattern work with RequestListener, though you may need to use a Handler to post the new load from onLoadFailed in RequestListener to avoid broken recursive state.
